packages/lib/src/components/ApplePay/ApplePay.tsx

```diff
@@ -58,33 +57,41 @@ class ApplePayElement extends UIElement<ApplePayElementProps> {
     }
 
     private startSession(onPaymentAuthorized: OnAuthorizedCallback) {
-        const { version, onValidateMerchant, onCancel, onPaymentMethodSelected, onShippingMethodSelected, onShippingContactSelected } = this.props;
-
-        return new Promise((resolve, reject) => this.props.onClick(resolve, reject)).then(() => {
-            const paymentRequest = preparePaymentRequest({
-                companyName: this.props.configuration.merchantName,
-                ...this.props
-            });
-
-            const session = new ApplePayService(paymentRequest, {
-                version,
-                onCancel,
-                onPaymentMethodSelected,
-                onShippingMethodSelected,
-                onShippingContactSelected,
-                onValidateMerchant: onValidateMerchant || this.validateMerchant,
-                onPaymentAuthorized: (resolve, reject, event) => {
-                    if (!!event.payment.token && !!event.payment.token.paymentData) {
-                        this.setState({ applePayToken: btoa(JSON.stringify(event.payment.token.paymentData)) });
-                    }
-
-                    super.submit();
-                    onPaymentAuthorized(resolve, reject, event);
-                }
-            });
+        const { version, onValidateMerchant, onPaymentMethodSelected, onShippingMethodSelected, onShippingContactSelected } = this.props;
 
-            session.begin();
+        const paymentRequest = preparePaymentRequest({
+            companyName: this.props.configuration.merchantName,
+            ...this.props
         });
+
+        const session = new ApplePayService(paymentRequest, {
```

Comment:
The conversation in issue #1801 talks about this being a breaking change - is it?

Reply:
I thought that it could be a breaking change by that time, but after performing the code change and some tests, it seems like that isn't the case. 

The callback is the same, and it uses the same parameters. The only difference is that we are creating the AppleSession before calling onClick, which doesn't affect the flow